threeML/plugins/OGIP/pha.py

```diff
@@ -543,8 +544,25 @@ def from_fits_file(cls,fits_file):
         with fits.open(fits_file) as f:
 
 
-            spectrum = FITSExtension.from_fits_file_extension(f['SPECTRUM'])
-
+            if 'SPECTRUM' in f: 
+                spectrum_extension=f['SPECTRUM']
+            else:
+                warnings.warn("unable to find SPECTRUM extension: not OGIP PHA!")
+                spectrum_extension=None
+                for extension in f:
```

Comment:
you can get more easily the value of a keyword using the .get() method of the header:

http://docs.astropy.org/en/stable/io/fits/api/headers.html

It returns the value if present, or None if not present. So for example:

```
if 'HDUCLASS' in extension.header and extension.header['HDUCLASS'] == 'OGIP'...
```
becomes:
```
if extension.header.get('HDUCLASS')=='OGIP'...
```
You can probably restructure this as:
```
for extension in f:
    hduclass = extension.header.get("HDUCLASS")
    hduclass1 = extension.header.get("HDUCLAS1")
    
    if hduclass == 'OGIP' and hduclas1 == 'SPECTRUM':
        # Found spectrum extension
        spectrum_extension = extension
        # Issue warning
        warnings.warn("File has no SPECTRUM extension, but found a spectrum in extension %s" % (spectrum_extension.header.get("EXTNAME")))
        # Reset name to SPECTRUM
        spectrum_extension.header['EXTNAME'] = 'SPECTRUM'
        # No need to continue looping
        break
```

(I didn't test the above code, there might be some small problem)